internal/config/config.go

```diff
@@ -310,6 +320,41 @@ func setClowderConfiguration(v *viper.Viper, clowderConfig *clowder.AppConfig) {
 	// Metrics configuration
 	v.Set("metrics.path", clowderConfig.MetricsPath)
 	v.Set("metrics.port", clowderConfig.MetricsPort)
+
+	// Set TLS configuration
+	processPublicEndpoint("rbac", "v1", v, clowderConfig)
+}
+
+func processPublicEndpoint(serviceName, version string, v *viper.Viper, clowderConfig *clowder.AppConfig) {
```

Comment:
From the method signature, I did not get that it sets the configuration value in `v` based on the service name. I.e. it is a bit cryptic and I needed to read it whole to understand it. The cause might be that the function is solving multiple problems. 

I'm thinking of approach:

```go

func getEndpointPath(serviceName, version String, clowderConfig *clowder.AppConfig) String {
   // most of the code here

   // here the question is whether is should also return a non-TLS path if the service doesn't have TLS path
}

func updateServiceBasePath(serviceName, version string, v *viper.Viper, clowderConfig *clowder.AppConfig) {
    basePath := getEndpointPath(serviceName, version, v, clowderConfig)
    if basePath != "" {
        // Q is whether this should rather be a param to the updateServiceBasePath function
        // and we could skip the "calculation". 
        configPath := getServiceBaseURLConfigPath(serviceName, version)  
        slog.Debug("base url for service " + serviceName + "  updated to '" + basePath + "'")
		v.Set(configPath, basePath)
    }
}

```